Merge autoconfiguration modules

larsgrefer opened the pull request:
see #255 

closes #255

persapiens commented:
Software structure is defined [here](https://github.com/joinfaces/joinfaces/wiki/Structure-of-the-software).

I had to create specific modules to tomcat, jetty and undertow because of [war packaging](https://github.com/joinfaces/joinfaces/issues/77) some time ago. I'll investigate again and report here.

Integration module was created to solve some integration technologies between JSF and Spring Boot.   For example, [CDI integration should be redesigned](https://github.com/joinfaces/joinfaces/issues/187).

larsgrefer commented:
Hi @persapiens ,

I'm aware of how the software is structured, but not exactly why its structured this way.

The common spring boot way is having all the auto-configuration code in one jar and only the relevant configuration is activated with `@ConditionalOnXXX` annotations.
So merging the `tomcat-`/`jetty-`/`undertow-autoconfigure` modules into the main `-autoconfigure`-module seems to be no problem. (Works well in this PR)

The `-integration`-module is a bit more complicated because it contains different things:
- a BeanFactoryPostProcessor (+ its auto-configuration code) for 
  - the view scope and 
  - supporting JSF and CDI Annotations
- the spring-security taglib
- unit test code for mocking a JSF environment.

I reorganized the `-integration`-module in the following steps
1. I moved everything in the `org.joinfaces.annotation`-package to the main `-autoconfigure`-module, because its "just" glue-code between jsf and spring.
2. Doing this arises the problem that both the `-autoconfigure` and the `-integration`-module need the same base classes in its unit tests, so I created an additional `test-utils`-module to share the code for mocking a JSF environment.
3. After that only the spring-security taglib remained in the `-integration` module so I renamed it to `jsf-spring-security-taglib`

larsgrefer commented:
I'll split this into smaller PR's
